### PR TITLE
do not try and compute variations of developer measures

### PR DESCRIPTION
**Pull request by sns-seb**
this fix bug on Dory where variations are no more computed for NCLOC, ...

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/ReportComputeMeasureVariationsStepTest.java [hunk 1]
```diff
@@ -226,6 +221,28 @@ public void set_variation_on_all_numeric_metrics() {
     assertThat(measureRepository.getRawMeasure(PROJECT, BUILD_BREAKER_METRIC).get().getVariations().getVariation1()).isEqualTo(-1d);
   }
 
+  @Test
+  public void do_not_set_variations_on_numeric_metric_for_developer() {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/ReportComputeMeasureVariationsStepTest.java]: This test is wrong : even without the modification of the prod code it's green !
You should do something like : 
`````
    SnapshotDto period1ProjectSnapshot = newSnapshotForProject(PROJECT_DTO);
    dbClient.snapshotDao().insert(session, period1ProjectSnapshot);
    dbClient.measureDao().insert(session,
      newMeasureDto(ISSUES_METRIC.getId(), PROJECT_DTO.getId(), period1ProjectSnapshot.getId(), 60d),
      newMeasureDto(ISSUES_METRIC.getId(), PROJECT_DTO.getId(), period1ProjectSnapshot.getId(), 60d).setDeveloperId(10L)
      );
    session.commit();

    periodsHolder.setPeriods(newPeriod(1, period1ProjectSnapshot));

    treeRootHolder.setRoot(PROJECT);

    addRawMeasure(PROJECT, ISSUES_METRIC, Measure.newMeasureBuilder().create(80, null));
    addRawMeasure(PROJECT, ISSUES_METRIC, Measure.newMeasureBuilder().forDeveloper(new DumbDeveloper("DEV1")).create(80, null));

    underTest.execute();

    assertThat(measureRepository.getRawMeasure(PROJECT, ISSUES_METRIC).get().getVariations().getVariation1()).isEqualTo(20d);
`````

You'll also need to update MeasureRepositoryRule.addRawMeasure() to allow dev measures : 
`````
checkAndInitProvidersState();

    InternalKey internalKey = new InternalKey(getRef(componentProvider.getByRef(componentRef)), metricKey, measure.getRuleId(),
      measure.getCharacteristicId(), measure.getDeveloper());
    checkState(!rawMeasures.containsKey(internalKey), format(
      "A measure can only be set once for Component (ref=%s), Metric (key=%s), ruleId=%s, characteristicId=%s",
      componentRef, metricKey, measure.getRuleId(), measure.getCharacteristicId()));

    rawMeasures.put(internalKey, measure);
    initialRawMeasures.put(internalKey, measure);

    return this;
`````

**Comment by julienlancelot** [comment]
Please fix the UT and LGTM

**Comment by julienlancelot** [comment]
And don't forget to associate the commit to SONAR-7027

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/ReportComputeMeasureVariationsStepTest.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/ReportComputeMeasureVariationsStepTest.java]: I did something of the like.
thank you a lot for spotting the test was crap, I also improved production code

**Pull request merged**